modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm

```diff
@@ -985,10 +985,129 @@
 //Donation reward for GoldenAlpharex
 /obj/item/clothing/glasses/welding/goldengoggles
 	name = "steampunk goggles"
-	desc = "This really feels like something you'd expect to see sitting on top of a certain ginger's head..."
+	desc = "This really feels like something you'd expect to see sitting on top of a certain ginger's head... They have a rather fancy brass trim around the lenses."
 	icon = 'modular_skyrat/master_files/icons/donator/obj/clothing/glasses.dmi'
 	worn_icon = 'modular_skyrat/master_files/icons/donator/mob/clothing/eyes.dmi'
 	icon_state = "goldengoggles"
+	slot_flags = ITEM_SLOT_EYES | ITEM_SLOT_HEAD // Making it fit in both first just so it can properly fit on the head slot in the loadout
 	flash_protect = FLASH_PROTECTION_NONE
+	flags_cover = GLASSESCOVERSEYES
+	custom_materials = null // Don't want that to go in the autolathe
 	visor_vars_to_toggle = 0
 	tint = 0
+	/// Was welding protection added yet?
+	var/welding_upgraded = FALSE
+	/// Was welding protection toggled on, if welding_upgraded is TRUE?
+	var/welding_protection = FALSE
+	/// The sound played when toggling the shutters.
+	var/shutters_sound = 'sound/effects/clock_tick.ogg'
+
+/obj/item/clothing/glasses/welding/goldengoggles/Initialize()
+	. = ..()
+	visor_toggling()
+
+/obj/item/clothing/glasses/welding/goldengoggles/examine(mob/user)
+	. = ..()
+	if(welding_upgraded)
+		. += "It has been upgraded with welding shutters, which are currently [welding_protection ? "closed" : "opened"]."
+
+/obj/item/clothing/glasses/welding/goldengoggles/item_action_slot_check(slot, mob/user)
+	. = ..()
+	if(. && slot == ITEM_SLOT_HEAD)
+		return FALSE
+
+/obj/item/clothing/glasses/welding/goldengoggles/attack_self(mob/user)
+	if(user.get_item_by_slot(ITEM_SLOT_HEAD) == src)
+		to_chat(user, span_warning("You can't seem to slip those on your eyes from the top of your head!"))
+		return
+	. = ..()
+
+/obj/item/clothing/glasses/welding/goldengoggles/visor_toggling()
+	. = ..()
+	slot_flags = up ? ITEM_SLOT_EYES | ITEM_SLOT_HEAD : ITEM_SLOT_EYES
+	toggle_vision_effects()
+
+/obj/item/clothing/glasses/welding/goldengoggles/weldingvisortoggle(mob/user)
+	. = ..()
+	handle_sight_updating(user)
+
+/obj/item/clothing/glasses/welding/goldengoggles/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(!istype(attacking_item, /obj/item/clothing/glasses/welding))
+		..()
+	if(welding_upgraded)
+		to_chat(user, span_warning("\The [src] was already upgraded to have welding protection!"))
```

Comment:
```suggestion
		to_chat(user, span_warning("\The [src] were already upgraded to have welding protection!"))
```

could do `[src.p_were()]` but that feels like overkill

Reply:
Yeah, I don't plan on having them change quantity ;)

Reply:
You never know with subtypes, but yes it's unlikely